tests/lang/s05_pragmas/s02_misc/t03_hint_pragmas.nim

```diff
@@ -0,0 +1,42 @@
+discard """
+description: '''
+`{.hint.}` pragma test
+'''
+
+nimout: '''
+t03_hint_pragmas.nim(15, 7) Hint: User-provided hint message [User]
+t03_hint_pragmas.nim(19, 9) Hint: gen1 size of the argument is 8 [User]
+t03_hint_pragmas.nim(34, 6) Hint: 'declaredButNotUsed' is declared but not used [XDeclaredButNotUsed]
+'''
+
+
+"""
+
+{.hint: "User-provided hint message".}
+
+
+proc gen1[T](arg: T) =
+  {.hint: "gen1 size of the argument is " & $sizeof(arg).}
+
+gen1[int](1)
+
+when false:
+  proc gen2[T](arg: T) =
+    {.line: instantiationInfo().}:
+      {.hint: "gen2 size of the argument is " & $sizeof(arg).}
+
+  gen2[int](1)
+
+  # REVIEW - this does not work. It should work or not?
+  {.line: ("override-file.nim", 999, 999).}:
```

Comment:
I don't think these use the same data, the former will effect stack spaces because they change the generated code, while emitting the hint uses the AST and its location info